File: csmrb/examples/example_with_yaml_file.rb
require './lib/csm'

# Set the default color scheme
DEFAULT_COLORS = {
  'logo' => 'red',
  'text' => 'blue',
  'result' => 'light_white'
}.freeze

custom_settings_file = 'examples/custom_settings.yaml'

# Load custom colors from YAML file
custom_colors = ColorSchemeManager::load_custom_yaml_file(custom_settings_file)
puts "Loaded custom colors: #{custom_colors}"
all_colors = ColorSchemeManager::load_colors(DEFAULT_COLORS, custom_colors)

puts "All colors: #{all_colors}"
puts "This is a #{'logo'.send(all_colors['logo'])} example of #{'text'.send(all_colors['text'])} using #{'result'.send(all_colors['result'])} colors."

File: csmpy/examples/example_with_yaml_file.py
from csmpy import ColorSchemeManager

# Define default colors using color names
DEFAULT_COLORS = {
    'logo': 'red',
    'text': 'blue',
    'result': 'light_white',
}

# Load custom colors from a YAML file
custom_colors = ColorSchemeManager.load_custom_yaml_file('examples/custom_settings.yaml')

# Merge default and custom colors
all_colors = ColorSchemeManager.load_colors(DEFAULT_COLORS, custom_colors)

# Example usage
logo_color = all_colors['logo']
text_color = all_colors['text']
result_color = all_colors['result']

print(f"This is a {logo_color} example of {text_color} using {result_color} colors.")
